cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -71,3 +74,27 @@ func (r *Reporter) send(spans []*model.Span, process *model.Process) error {
 	}
 	return err
 }
+
+// addTags appends jaeger tags for the agent to every span it sends to the collector.
+func addProcessTags(spans []*model.Span, process *model.Process, agentTags []model.KeyValue) ([]*model.Span, *model.Process) {
+	if process != nil {
+		process.Tags = append(process.Tags, agentTags...)
+		return spans, process
+	}
+	for _, span := range spans {
+		if len(agentTags) > 0 {
```

Comment:
This check can be at the top of the function (return immediately if no agent tags).

Instead, here the check  should be `if span.Process != nil`